src/vs/workbench/api/common/extHostMemento.ts

```diff
@@ -51,7 +55,38 @@ export class ExtensionMemento implements vscode.Memento {
 
 	update(key: string, value: any): Promise<void> {
 		this._value![key] = value;
-		return this._storage.setValue(this._shared, this._id, this._value!);
+
+		let record = this._deferredPromises.get(key);
+		if (record !== undefined) {
+			return record.p;
+		}
+
+		const promise = new DeferredPromise<void>();
+		this._deferredPromises.set(key, promise);
+
+		if (this._scheduler) {
+			this._scheduler.cancel();
+		}
+
+		this._scheduler = new RunOnceScheduler(() => {
+			const records = this._deferredPromises;
+			this._deferredPromises = new Map();
+			(async () => {
+				try {
+					await this._storage.setValue(this._shared, this._id, this._value!);
+					for (value of records.values()) {
+						value.resolve();
+					}
+				} catch (e) {
+					for (value of records.entries()) {
+						value.resolve();
+					}
+				}
+			})();
+		}, 0);
+		this._scheduler.schedule();
```

Comment:
I think you can leave the scheduler unchanged, e.g initialise it globally (in the ctor) and then only schedule it if it wasn't already scheduled, there is `isScheduled` for that.